packages/ozone/src/set/service.ts

```diff
@@ -0,0 +1,203 @@
+import Database from '../db'
+import { Selectable, sql } from 'kysely'
+import { OzoneSet } from '../db/schema/ozone_set'
+import { SetView } from '../lexicon/types/tools/ozone/sets/defs'
+
+export type SetServiceCreator = (db: Database) => SetService
+
+export class SetService {
+  constructor(public db: Database) {}
+
+  static creator() {
+    return (db: Database) => new SetService(db)
+  }
+
+  async query({
+    limit,
+    cursor,
+    namePrefix,
+    sortBy,
+  }: {
+    limit: number
+    cursor?: string
+    namePrefix?: string
+    sortBy: 'name' | 'createdAt' | 'updatedAt'
+  }): Promise<{
+    sets: Selectable<OzoneSet & { setSize: number }>[]
+    cursor?: string
+  }> {
+    let qb = this.db.db
+      .selectFrom('ozone_set as s')
+      .leftJoin(
+        (eb) =>
+          eb
+            .selectFrom('ozone_set_value')
+            .select(['setId'])
+            .select((eb) => eb.fn.count('id').as('count'))
+            .groupBy('setId')
+            .as('sv'),
+        (join) => join.onRef('sv.setId', '=', 's.id'),
+      )
+      .select([
+        's.id',
+        's.name',
+        's.description',
+        's.createdAt',
+        's.updatedAt',
+        sql<number>`coalesce(sv.count, 0)`.as('setSize'),
+      ])
```

Comment:
You may be able to nix the complex join with something like:
```suggestion
      .select([
        's.id',
        's.name',
        's.description',
        's.createdAt',
        's.updatedAt',
        (eb) =>
          eb
            .selectFrom('ozone_set_value')
            .select((e) => e.fn.count('setId').as('count'))
            .whereRef('setId', '=', 's.id')
            .as('setSize'),
      ])
```